django/core/management/commands/test.py

```diff
@@ -18,11 +19,7 @@ def run_from_argv(self, argv):
         option. This allows a test runner to define additional command line
         arguments.
         """
-        option = '--testrunner='
-        for arg in argv[2:]:
-            if arg.startswith(option):
-                self.test_runner = arg[len(option):]
-                break
+        self.test_runner = preparse_commandline_option(argv, '--testrunner')
```

Comment:
`preparse_command_line_option` or maybe `get_command_line_option`?

Reply:
Thanks for your comments. I've improved the commit based on all of them. I added an extra testcase to test an invalid `--testrunner` option (no argument) to trigger the CommandError you pointed out, which should test this option. I could not confirm with coverage testing, I suspect because the test forks to test the command rather than calling it internally.

As for the structure of this fix, see also my comment at https://github.com/django/django/pull/10117#issuecomment-414124737, which describes a problem with this approach when calling management commands using `call_command` rather than on the commandline directly. Regardless, the commit in this PR is still an improvement, even if the approach will be later changed to fix the problem described there.